tsconfig.json

```diff
@@ -13,7 +13,16 @@
     "target": "ES2016",
     "baseUrl": ".",
     "lib": ["es2017", "es2019", "dom", "dom.iterable"],
-    "rootDirs": [".", "bazel-bin"]
+    "rootDirs": [
+      ".",
+      "bazel-bin",
+      "bazel-out/k8-opt/bin",
+      "bazel-out/k8-fastbuild/bin",
+      "bazel-out/darwin-opt/bin",
+      "bazel-out/darwin-fastbuild/bin",
+      "bazel-out/darwin_arm64-opt/bin",
+      "bazel-out/darwin_arm64-fastbuild/bin"
```

Comment:
I thought I remembered seeing something here that we were doing some other workaround thing so that we didn't have to list all of these in our tsconfig: https://github.com/buildbuddy-io/buildbuddy/pull/1956/files#diff-2085a400b69ac0ec4291b70e84b04a90a144e10b8cea4b4f082d70dce50ce015R11

Reply:
There was some issue I ran into when omitting these, and I meant to dig into it and see what the root issue was. I'll dig more into it next time I revisit this PR

Reply:
I removed these and it builds just fine -- maybe I needed these during some intermediate state as I was migrating things, but everything builds fine without these now.

Reply:
jk, the typecheck tests fail without these. I would like to dig into it at some point and see why these are needed, but would like to decouple that from this PR